ggshield/cmd/auth/logout.py

```diff
@@ -0,0 +1,104 @@
+import click
+from requests.exceptions import ConnectionError
+
+from ggshield.core.client import create_client
+from ggshield.core.config import Config
+from ggshield.core.utils import dashboard_to_api_url
+
+
+REVOKE_FAIL_MESSAGE = (
+    "Logout failed due to an error when revoking the token, "
+    "you can skip revocation with --no-revoke to bypass"
+)
+
+
+@click.command()
+@click.option(
+    "--instance",
+    required=False,
+    type=str,
+    help="URL of the instance to logout from.",
+)
+@click.option(
+    "--no-revoke",
+    is_flag=True,
+    help="Whether the token should be revoked before being removed from the config.",
+)
+@click.option(
+    "--all", "all_tokens", is_flag=True, help="Iterate over every saved tokens."
+)
+@click.pass_context
+def logout_cmd(
+    ctx: click.Context, instance: str, no_revoke: bool, all_tokens: int
+) -> int:
+    """
+    Delete local auth config for the specified instance (or default instance if not specified)
```

Comment:
"local auth config" sounds a bit too ggshield-internal to me.

```suggestion
    Delete saved authentication details for the specified instance (or default instance if not specified).
```